libbeat/template/load.go

```diff
@@ -143,50 +142,16 @@ func (l *ESLoader) loadTemplate(templateName string, templateType IndexTemplateT
 	return nil
 }
 
+// templateExists checks if a template exists
 func (l *ESLoader) templateExists(templateName string, templateType IndexTemplateType) (bool, error) {
-	if templateType == IndexTemplateComponent {
-		return l.checkExistsComponentTemplate(templateName)
-	}
-	return l.checkExistsTemplate(templateName)
-}
-
-// existsTemplate checks if a given template already exist, using the
-// `_cat/templates/<name>` API.
-//
-// An error is returned if the loader failed to execute the request, or a
-// status code indicating some problems is encountered.
-func (l *ESLoader) checkExistsTemplate(name string) (bool, error) {
-	status, body, err := l.client.Request("GET", "/_cat/templates/"+name, "", nil, nil)
-	if err != nil {
-		return false, err
-	}
-
-	// Elasticsearch API returns 200, even if the template does not exists. We
-	// need to validate the body to be sure the template is actually known. Any
-	// status code other than 200 will be treated as error.
-	if status != http.StatusOK {
-		return false, &StatusError{status: status}
-	}
-	return strings.Contains(string(body), name), nil
-}
-
-// existsComponentTemplate checks if a component template exists by querying
-// the `_component_template/<name>` API.
-//
-// The resource is assumed as present if a 200 OK status is returned and missing if a 404 is returned.
-// Other status codes or IO errors during the request are reported as error.
-func (l *ESLoader) checkExistsComponentTemplate(name string) (bool, error) {
-	status, _, err := l.client.Request("GET", "/_component_template/"+name, "", nil, nil)
-
+	path := templateLoaderPath[templateType] + templateName
```

Comment:
should we proceed if templateLoaderPath does not have a templateType entry? 
we can save turnaround by a simple check

Reply:
In theory, templateType is always a valid type, but I will add a check. Better to be safe than sorry.